.github/workflows/ci.yml

```diff
@@ -1,11 +1,16 @@
 name: CI
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```

Comment:
Does this run CI pipeline for new pull requests ?

Reply:
it run the ci only if the PR is opened against master and when we merge/push to master too.
This will allow us to push in other branches and not have the ci triggered and the email of failure dispatched unless we open a PR.